### ISSUE TITLE
Allow tasks.json to generate multiline problem messages

### ISSUE THREAD

**Issue by AshleyYakeley** [Microsoft/vscode#69183]
So 1.31 apparently introduces multiline messages in the problem panel. But there does not seem to be any documented way to actually _generate_ these messages, such as in `tasks.json`, from arbitrary build tools.

The available documentation mentions "multiline problems", but the messages must still be each on a single line, whereas the messages I want to match are each multiline. Here's an example, from the output of the GHC compiler:

```
    /home/ashley/Projects/Haskell/Truth/truth-gtk/text/Main.hs:62:44: error:
        • Couldn't match expected type ‘(UISpec sel edit, Bool)
                                        -> (UISpec sel edit, Bool)’
                      with actual type ‘(UISpec sel1 tedit1, Bool)’
        • The function ‘(menuBarUISpec mbar, False)’
          is applied to one argument,
          but its type ‘(UISpec sel1 tedit1, Bool)’ has none
          In the expression: (menuBarUISpec mbar, False) (spec, True)
          In the first argument of ‘verticalUISpec’, namely
            ‘[(menuBarUISpec mbar, False) (spec, True)]’
        • Relevant bindings include
            spec :: UISpec sel edit (bound at text/Main.hs:59:50)
            simpleUI :: (UIWindow, ()) -> UISpec sel edit -> UISpec sel edit
              (bound at text/Main.hs:59:21)
       |
    62 |                         in verticalUISpec [(menuBarUISpec mbar,False)(spec,True)]
       |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    
```

Currently the best I can do is match the first line.

**Comment by alexr00** [comment]
Info about multiline problem matchers for tasks is here: https://code.visualstudio.com/Docs/editor/tasks#_defining-a-multiline-problem-matcher

**Comment by AshleyYakeley** [comment]
This is not correct!! As I pointed out:

**The available documentation mentions "multiline problems", but the messages must still be each on a single line, whereas the messages I want to match are each multiline.**

**Comment by AshleyYakeley** [comment]
@alexr00 just to clarify, the documentation you point to is for multiline problems with **single-line messages**. The feature introduced in 1.31 is multiline problems with **multi-line messages**.

**Comment by alexr00** [comment]
I see. Then this is a duplicate of #9635, which we are considering adding API for.

**Comment by AshleyYakeley** [comment]
OK, it looks like you're treating the change you made in 1.31 as a partial fix to #9635, leaving that issue open until it's complete.

Thanks!
